File: mixed/SimpleVM.java
import java.util.Stack;

public class SimpleVM {
    private Stack<Integer> stack = new Stack<>(); // Stack to store values

    // Method to execute a set of instructions
    public void execute(String[] instructions) {
        for (String instr : instructions) {
            String[] parts = instr.split(" "); // Split instruction into parts
            switch (parts[0]) {
                case "PUSH":
                    // Push the given number onto the stack
                    stack.push(Integer.parseInt(parts[1]));
                    break;
                case "ADD":
                    // Pop the top two values from the stack, add them, and push the result
                    stack.push(stack.pop() + stack.pop());
                    break;
                case "PRINT":
                    // Print the top value of the stack
                    System.out.println(stack.peek());
                    break;
                default:
                    // Handle invalid instructions
                    System.out.println("Invalid instruction: " + instr);
            }
        }
    }

    public static void main(String[] args) {
        // Create an instance of the SimpleVM
        SimpleVM vm = new SimpleVM();
        
        // Define a sequence of instructions
        String[] program = {
            "PUSH 5",  // Push 5 onto the stack
            "PUSH 3",  // Push 3 onto the stack
            "ADD",      // Add the top two values (5 + 3)
            "PRINT"     // Print the result (8)
        };
        
        // Execute the program
        vm.execute(program);
    }
}

File: interpreted/simple_vm.py
class SimpleVM:
    def __init__(self):
        self.stack = []  # Stack to store values

    def execute(self, instructions):
        for instr in instructions:
            parts = instr.split()
            if parts[0] == "PUSH":
                self.stack.append(int(parts[1]))  # Push a number onto the stack
            elif parts[0] == "ADD":
                self.stack.append(self.stack.pop() + self.stack.pop())  # Pop two values, add them, and push the result
            elif parts[0] == "PRINT":
                print(self.stack[-1])  # Print the top value of the stack
            else:
                print(f"Invalid instruction: {instr}")  # Handle unknown instructions

if __name__ == "__main__":
    vm = SimpleVM()
    program = [
        "PUSH 5",  # Push 5 onto the stack
        "PUSH 3",  # Push 3 onto the stack
        "ADD",      # Add top two values (5 + 3)
        "PRINT"     # Print the result (8)
    ]
    vm.execute(program)
